trieTools/zeroBalanceSystemAccountChecker/tokenBalanceGetter.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+	"strings"
+
+	"github.com/tidwall/gjson"
+)
+
+type tokenBalanceGetter struct {
+	proxyURL string
+}
+
+func newTokenBalanceGetter(proxyURL string) *tokenBalanceGetter {
+	return &tokenBalanceGetter{
+		proxyURL: proxyURL,
+	}
+}
+
+func (tbg *tokenBalanceGetter) getBalance(address, token string) (string, error) {
+	tokenID, nonce := getTokenIDAndNonce(token)
+	return tbg.fetchTokenBalanceWithRetrial(address, tokenID, nonce)
+}
+
+func getTokenIDAndNonce(token string) (string, uint64) {
+	tokens := strings.Split(token, "-")
+	tokenID := tokens[0] + "-" + tokens[1]
+
+	nonce := big.NewInt(0)
+	nonce.SetString(tokens[2], 16)
+
+	return tokenID, nonce.Uint64()
+}
+
+func (tbg *tokenBalanceGetter) fetchTokenBalanceWithRetrial(address, tokenID string, nonce uint64) (string, error) {
+	ctRetrials := 0
+
+	for ctRetrials < maxRequestsRetrial {
+		url := fmt.Sprintf("%s/address/%s/nft/%s/nonce/%d", tbg.proxyURL, address, tokenID, nonce)
+		resp, errHttp := http.Get(url)
```

Comment:
hard to test. Why didn't you used an interface here?

Reply:
Replaced with pointer to func